### PR TITLE
#159987403 social media authentication

### PR DESCRIPTION
**Pull request by IyiKuyoro**
#### What does this PR do?
Implement social media authentication
#### Description of Task to be completed?
- Install passport and passport strategies
- setup authentication with facebook
- setup authentication with google
#### How should this be manually tested?
1. Clone the repository.
2. Checkout the `ft-social-login-#159987403` branch.
3. Run npm install on your local machine.
4. Run the db migrations using `npm run migrate`.
5. Start the server using `npm run start_dev`.
5. Access the following routes on a browser; `localhost:8000/api/users/auth/google` and `localhost:8000/api/users/auth/facebook`.
#### Any background context you want to provide?
This feature should enable users to signup and signin using their social media accounts on google and facebook
#### What are the relevant pivotal tracker stories?
#159987403

### CODE REVIEW

File: server/config/passportSetup.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+passport.deserializeUser((id, done) => {
+  Users.findById(id).then((user) => {
+    done(null, user);
+  });
+});
+
+passport.use(
+  new GoogleStrategy({
+    callbackURL: 'http://localhost:8000/api/users/auth/google/redirect',
```
Comment (iAmao) [inline server/config/passportSetup.js]: Add this to env variable

**Review by iAmao — CHANGES_REQUESTED** [review]
Nice one again 👍

**Comment by IyiKuyoro** [comment]
@iAmao resolved all issues and implemented a mock test that mocks the facebook and google authentication call

**Comment by IyiKuyoro** [comment]
@iAmao  This PR is ready to merge

File: test/test.spec.js [hunk 2]
```diff
@@ -215,6 +216,44 @@ describe('TEST ALL ENDPOINT', () => {
...
+          user: {
+            id: 1,
+            firstname: 'Opeoluwa',
+            lastname: 'Iyi-Kuyoro',
+            email: 'op@ah.metis.com',
+            created: true
+          }
+        })
+        .end((err, res) => {
+          // console.log(res);
```
Comment (iAmao) [inline test/test.spec.js]: Remove commented code

File: test/test.spec.js [hunk 3]
```diff
@@ -215,6 +216,44 @@ describe('TEST ALL ENDPOINT', () => {
...
```
Comment (iAmao) [inline test/test.spec.js]: Remove commented code

**Review by iAmao — CHANGES_REQUESTED** [review]
Nice, just fix the minors

**Review by iAmao — APPROVED** [review]

**Pull request merged**